src/frontend/Ast_to_Mir.ml

```diff
@@ -49,52 +59,55 @@ and trans_expr {Ast.expr; Ast.emeta} =
         | ArrayExpr eles ->
             FunApp
               ( CompilerInternal
-              , string_of_internal_fn FnMakeArray
+              , Internal_fun.to_string FnMakeArray
               , trans_exprs eles )
         | RowVectorExpr eles ->
             FunApp
               ( CompilerInternal
-              , string_of_internal_fn FnMakeRowVec
+              , Internal_fun.to_string FnMakeRowVec
               , trans_exprs eles )
         | Indexed (lhs, indices) ->
             Indexed (trans_expr lhs, List.map ~f:trans_idx indices)
         | Paren _ | BinOp _ | PrefixOp _ | PostfixOp _ ->
             raise_s [%message "Impossible!"]
-      in
-      {expr; emeta= {mtype; mloc; madlevel}}
+      and meta = Expr.Typed.Meta.create ~type_ ~adlevel ~loc () in
+      Expr.Fixed.fix (meta, pattern)
```

Comment:
Could this be also written as 
`Expr.Fixed.{meta={Typed.Meta.type_; adlevel; loc}; pattern}`?

Reply:
Almost, it would be `Expr.{  Typed.meta = Meta.{type_;adlevel;loc}; pattern }`.